packages/cli-kit/src/node/cli.ts

```diff
@@ -52,4 +54,50 @@ export async function runCreateCLI(options: RunCLIOptions) {
   await runCLI(options)
 }
 
+export async function replaceGlobalCLIWithLocal(filepath: string): Promise<boolean> {
+  // Temporary flag while we test out this feature and ensure it won't break anything!
+  if (!isTruthy(process.env[constants.environmentVariables.enableCliRedirect])) return false
+  if (isTruthy(process.env[constants.environmentVariables.skipCliRedirect])) return false
+  if (process.env.npm_config_user_agent) return false
```

Comment:
The connection between the `npm_config_user_agent` environment variable and the returned value is not obvious. What about adding an explanatory comment?